nautobot/core/celery/__init__.py

```diff
@@ -39,6 +43,28 @@
 app.autodiscover_tasks()
 
 
+@signals.worker_ready.connect()
+def setup_prometheus(**kwargs):
+    """This sets up an HTTP server to serve prometheus metrics from the celery workers."""
+    # Don't set up the server if the port is undefined
+    if not settings.CELERY_WORKER_PROMETHEUS_PORT:
+        return
+
+    logger.info("Setting up prometheus metrics HTTP server for celery worker.")
+
+    # Ensure that the multiprocess coordination directory exists
+    multiprocess_coordination_directory = Path(os.environ["prometheus_multiproc_dir"])
+    multiprocess_coordination_directory.mkdir(parents=True, exist_ok=True)
+    # Delete any files lefts in the multiprocess coordination directory
+    for filepath in multiprocess_coordination_directory.glob("*.db"):
+        filepath.unlink()
```

Comment:
Any resource contention we would need to worry about if there are multiple workers sharing a filesystem, or if the worker shares a filesystem with the core app?

Reply:
That is an excellent point. I included this as part of a snippet in an [issue](https://github.com/korfuri/django-prometheus/issues/339) that details this problem - if it doesn't cause any problems we should absolutely drop these two lines.

Reply:
Changed to not delete files and added a comment.